pkg/monitor/cluster/cluster.go

```diff
@@ -67,6 +67,11 @@ func NewMonitor(env env.Interface, log *logrus.Entry, oc *api.OpenShiftCluster,
 func (mon *Monitor) Monitor(ctx context.Context) error {
 	mon.log.Debug("monitoring")
 
+	err := mon.emitClusterVersion()
```

Comment:
Hi @m1kola I was wondering about this myself, I checked the v3 code base and on the first glance it seems like we do use it in many places without ctx, eg. https://github.com/openshift/openshift-azure/blob/master/pkg/cluster/kubeclient/drain.go#L44-L54
Most functions that we use from CoreV1 are not taking ctx, isn't it accounted for in some other way by the package itself or other mechanisms?

Reply:
Hey. I'm not sure about v3, but in few places in v4 it's important. For example, in API we must have cancellation because ARM has a timeout of 30 seconds for a request, so if some request from RP API to API server hangs, we we will be wasting resources.

In case of minotoring, I think it's also important: we don't want to create a lot of slow requests. Currently the code that calls this function has 30 seconds timeout which is not being respected:

https://github.com/Azure/ARO-RP/blob/9f6883df71f9178ba948c629c0633225551c6499/pkg/monitor/worker.go#L153-L164

Reply:
`restConfig.Wrap(...)` hack is.. a hack and I don't like it. But it seems to be the clinest hack we can use while waiting for support of context to arrive in the `client-go`. 

Note that the rest client does support context. For example below `emitAPIServerHealthzCode` takes context because it uses rest client directly.

https://github.com/Azure/ARO-RP/blob/9f6883df71f9178ba948c629c0633225551c6499/pkg/monitor/cluster/healthz.go#L11-L23

Reply:
I agree @m1kola, keeping context is important. How haven't we stumbled on this earlier? 
this was recently implemented in upstream: https://github.com/kubernetes/client-go/commit/e7a922c979d0f5cd6131039b2c86af97a164c7e4 but I don't know when it is available in openshift.

Reply:
Pulled in Olga's commits that fix this per this morning's discussion.